file/file.go

```diff
@@ -111,3 +116,58 @@ func (t *fileTopic) Close() error {
 	}
 	return nil
 }
+
+// --------------------------------------------------------------------
+
+type fileSub struct {
+	root string
+}
+
+// NewSubscriber inits a subscriber within a root directory.
+// It assumes, that file is not being written to anymore.
+// It iterates entire file (all records) without tracking processed messages.
+func NewSubscriber(root string) bps.Subscriber {
+	return &fileSub{
+		root: root,
+	}
+}
+
+func (s *fileSub) Subscribe(ctx context.Context, topic string, handler bps.Handler) error {
+	name := filepath.Join(s.root, topic)
+	f, err := os.Open(name)
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	dec := json.NewDecoder(f)
+	for {
```

Comment:
`for dec.More() {`

Reply:
I like it, but current approach is cleaner in terms of ctx checking (if ctx is cancelled, it will return `context.Canceled` error; with `dec.More()`  it'll return `nil` if file is empty).

Don't think I should bother with this though.